superset-frontend/plugins/plugin-chart-cartodiagram/src/components/ChartLayer.tsx

```diff
@@ -228,14 +214,20 @@ export class ChartLayer extends Layer {
     const { viewState } = frameState;
     const currentZoom = Math.round(viewState.zoom);
 
-    let charts = this.chartsPerZoom[currentZoom];
-    // only create charts if they do not exist yet
-    if (!charts) {
-      this.createChartsPerZoom(currentZoom);
-      charts = this.chartsPerZoom[currentZoom];
+    // nextResolution is only defined while an animation
+    // is in action. For this time we show a loading mask
+    // to keep the amount of chart rerenderings as low as possible.
+    if (viewState.nextResolution !== undefined) {
+      return this.loadingMask;
```

Comment:
the loading animation takes quite long for me. Also the loading gif can be grabbed. Maybe there is a simple way to prevent any interaction from the user with it.

![Peek 2023-01-05 16-29](https://user-images.githubusercontent.com/15704517/210817569-4ec61149-65ec-495f-a213-e7705486d775.gif)

Reply:
Good catch. solved in https://github.com/terrestris/superset/pull/6/commits/46ab44fe2f00a857e1c0c55604cff748aa1e54be